lib/daru/dataframe.rb

```diff
@@ -1438,12 +1438,26 @@ def sort! vector_order, opts={}
         type: :quick_sort,
         by: {}
       }.merge(opts)
+      
+      def lol(r, vector_order, opts)
+        x = vector_order.map { |i| x = opts[:by][i]; !!x ? x.call(row[r][i]) : row[r][i] } if !!opts[:by]
+        (0..vector_order.size-1).each { |i| x[i] = -x[i] if !opts[:ascending][i] } if opts[:ascending].is_a? Array
+        x
+      end
 
-      opts[:by]        = create_logic_blocks vector_order, opts[:by]
+      # opts[:by]        = create_logic_blocks vector_order, opts[:by]
       opts[:ascending] = sort_order_array vector_order, opts[:ascending]
       idx = @index.to_a
-      send(opts[:type], vector_order, idx, opts[:by], opts[:ascending])
+      # send(opts[:type], vector_order, idx, opts[:by], opts[:ascending])
+      idx = @index.to_a.sort_by { |r| lol(r, vector_order, opts) }
       self.index = Daru::Index.new(idx)
+      # Following three lines are very slow 
+      # Master -> 54, 53, 51
+      # This branch (with following lines) -> 12, 13, 5
+      # This branch (without following lines) -> .03, .17, .06
+      for v in self.vectors
```

Comment:
I am sorry I just saw it now. 

Yes, in the latest implementation only positional index get sorted and then all vectors are reconstructed.